### PR TITLE
fix(github): update pulls.rs for octocrab 0.53 Option-wrapped fields

### LINKED ISSUE #1306: fix(github): update pulls.rs for octocrab 0.53 Option-wrapped fields
## Summary

`crates/aptu-core/src/github/pulls.rs` fails to compile against octocrab 0.53: 17 diagnostics (16 unique sites) because octocrab 0.53 wrapped previously-bare fields in `Option`.

The codebase compiles against no other crate until this is fixed.

## Root cause

octocrab 0.53 changed the following field types on `PullRequest`:

| Field | Old type | New type |
|---|---|---|
| `head` / `base` | `Head` / `Base` | `Option<Box<Head>>` / `Option<Box<Base>>` |
| `labels` | `Vec<Label>` | `Option<Vec<Label>>` |
| `title` | `String` | `Option<String>` |
| `number` | `u64` | `Option<u64>` |
| `html_url` | `Url` | `Option<Url>` |
| `additions` / `deletions` / `changed_files` | `u64` | `Option<u64>` |

## Fix

All 16 sites in `pulls.rs` are safe unwrap-with-default patterns. Required changes:

```rust
// head/base fields (5 sites for head, 2 for base)
pr.head.sha                  -> pr.head.as_deref().map(|h| h.sha.as_str()).unwrap_or_default()
pr.head.ref_field            -> pr.head.as_deref().map(|h| h.ref_field.clone()).unwrap_or_default()
pr.base.ref_field            -> pr.base.as_deref().map(|b| b.ref_field.clone()).unwrap_or_default()

// labels (1 site)
pr.labels.iter().map(|l| l.name.clone())
  -> pr.labels.iter().flat_map(|v| v.iter().map(|l| l.name.clone()))

// scalar Option fields
pr.title.clone()             -> pr.title.clone().unwrap_or_default()   // 2 sites
pr.html_url.to_string()      -> pr.html_url.as_ref().map(|u| u.to_string()).unwrap_or_default()  // 2 sites
pr.number                    -> pr.number.unwrap_or(0)                 // 1 site (display only)
pr.additions / pr.deletions  -> .unwrap_or(0)                          // 2 sites
pr.changed_files             -> pr.changed_files.map(|n| u32::try_from(n).unwrap_or(u32::MAX)).unwrap_or(0)  // 1 site
```

**Validated safe:**
- `as_deref()` on `Option<Box<T>>` is correct: `Box<T>: Deref<Target=T>` yields `Option<&T>`
- `PrDetails` struct fields are all `String` -- `unwrap_or_default()` matches
- Empty SHA path: `post_pr_review` guards `if !commit_id.is_empty()` before including it in payload
- `pr.number.unwrap_or(0)` blast radius is cosmetic only: `PrCreateResult.pr_number` is used for display, not API calls
- No other files affected (verified with `cargo check`)

## Files

- `crates/aptu-core/src/github/pulls.rs` (16 sites across ~lines 172, 189, 210, 216, 218–222, 693–701)

## Acceptance criteria

- [ ] `cargo check` exits 0
- [ ] `cargo test --no-run` exits 0
- [ ] `cargo clippy -- -D warnings` exits 0
- [ ] All 16 sites updated with the patterns above
- [ ] No changes outside `pulls.rs`

### PR DESCRIPTION
**Pull request by clouatre**
## Summary

Update `pulls.rs` and `pr.rs` for octocrab 0.53 breaking changes: PullRequest fields that were
previously non-Option are now Option-wrapped. All affected sites use safe unwrap-with-default
patterns. The `run_queue` path was further hardened to skip queue entries whose PR number resolves
to 0, emitting a `tracing::warn!` rather than silently corrupting output.

No public API changes. No behavioral changes for well-formed API responses.

## Changes

- `crates/aptu-core/src/github/pulls.rs`: Updated 16 call sites to handle Option-wrapped fields
  (`head`, `base`, `labels`, `title`, `html_url`, `number`, `changed_files`, `additions`,
  `deletions`) with `unwrap_or_default` patterns.
- `crates/aptu-cli/src/commands/pr.rs`: Fixed octocrab 0.53 breakage in `run_queue`; replaced
  `.map()` with `.filter_map()` so PRs with a zero or missing number are skipped with a warning
  instead of defaulting to 0 and corrupting queue output. Added `tracing::debug!` for mapped
  fields and a `test_queue_skips_pr_with_zero_number` edge-case test.

## Test plan

- [x] Tests pass (523 passed, 0 failed)
- [x] Linter clean (`cargo clippy -- -D warnings`)
- [x] Formatter clean (`cargo fmt --check`)
- [x] Security scan clean (no secrets, credentials, or PII)

Closes #1306

### CODE REVIEW

**Review by clouatre — COMMENTED** [review]
test

File: crates/aptu-cli/src/commands/pr.rs [hunk 1]
```diff
@@ -411,22 +412,44 @@ pub async fn run_queue(
 
     let mut queued_prs: Vec<crate::output::pr::QueuedPr> = full_prs
         .into_iter()
-        .map(|pr| {
+        .filter_map(|pr| {
+            let number = pr.number.unwrap_or(0);
+            if number == 0 {
```
[the remark this answers is not in the document]
Comment (clouatre) [reply crates/aptu-cli/src/commands/pr.rs:417]: Agreed. The `filter_map` pattern is forward-compatible: whether octocrab wraps `number` in an `Option` or keeps it as `u64`, the guard (skip on 0 or None) remains valid. If octocrab ever changes the type to `Option<u64>`, a single `.unwrap_or(0)` in the condition keeps the logic identical. Added a code comment at the check site to document this invariant.
